### ISSUE TITLE
How to specify interface for defaultGateway6

### ISSUE THREAD

**Issue by enko** [NixOS/nixpkgs#12500]
I'm using a ipv6-linklocal address as my default gateway, but it seems to me I cant specify the device for defaultGateway6?

I am doing something like this:

```
ip -6 route add default via fe80::5054:ff:f39f:a314 dev eth0
```

**Comment by globin** [comment]
We're using a workaround for something similar:
https://github.com/mayflower/mf-nixops/blob/1187c7ec7d530cd7e0701ea8c64751bbb26495fa/production/raumselektor.nix#L27-L29

I think @fadenb started working on a fix.

**Comment by fadenb** [comment]
I am not yet sure how to fix this issue in a clean way.
I was thinking about adding a new device route functionality to properly configure how to reach the gateway. Unfortunately this will not help with the link local issue. 
Any input on this is appreciated.

**Comment by Baughn** [comment]
@fadenb Are you stuck on what the configuration interface should be, or how to implement it?

**Comment by fpletz** [comment]
I think a general `network.routes` option providing most of `ip route` options would be appropriate. I needed this a few times already and settled with `networking.localCommands` hacks.

And while we're at it, `networking.rules` for policy routing and support for multiple routing tables would be cool, too. ;)

**Comment by fpletz** [comment]
This has been fixed by 0cb487ee044bbe5cfa1d6421df385b6bc2447bd2.
